CONSOLE-3113: Support ordering of plugin nav sections in admin perspective

TheRealJon opened the pull request:
- Migrate admin nav to dynamic plugin extensions
- Refactor nav components to remove admin nav special case
- Address some tech debt and code quality issues

TheRealJon commented:
/woof

glekner commented:
/retest

andrewballantyne reviewed (changes requested):
Is there anyway we can move the nav into `console-app`? Framework for building the navigation sounds very `console-app`.

TheRealJon commented:
@andrewballantyne I've addressed the easiest of your comments, but haven't moved the nav components into the console-app package yet. That's going to involve a little bit of refactoring to address some cyclical imports.

andrewballantyne commented:
> @andrewballantyne I've addressed the easiest of your comments, but haven't moved the nav components into the console-app package yet. That's going to involve a little bit of refactoring to address some cyclical imports.

@TheRealJon We don't **have to** move them with this work, but we always need to ask the natural follow up question... "if not now, when?", so I just asked it of you here. Let me know if you think it's too much.

As for your cyclical imports -- that's likely the curse of our `index` files. If you're hitting these issues, go directly to the file instead of stopping at an index file. Usually that will solve your issue.

TheRealJon commented:
QE Approver
/assign @yapei

Docs Approver:
/assign @opayne1

PX Approver:
/assign @RickJWagner

TheRealJon commented:
@andrewballantyne I think for the sake of moving this story along, we should try to get this merged and create a follow-on story to move the nav component into console-app.

RickJWagner commented:
/label px-approved

andrewballantyne commented on frontend/packages/knative-plugin/console-extensions.json:
```diff
@@ -1675,5 +1675,19 @@
+    "properties": {
+      "perspective": "admin",
+      "id": "serverless",
+      "name": "%public~Serverless%",
```
here

andrewballantyne commented on frontend/packages/kubevirt-plugin/console-extensions.json:
```diff
@@ -749,5 +749,19 @@
+    "properties": {
+      "perspective": "admin",
+      "id": "virtualization",
+      "name": "%public~Virtualization%",
```
here

andrewballantyne commented on frontend/packages/pipelines-plugin/console-extensions.json:
```diff
@@ -784,5 +784,22 @@
+    "properties": {
+      "perspective": "admin",
+      "id": "pipelines",
+      "name": "%public~Pipelines%",
```
here

andrewballantyne reviewed (changes requested):
Can you also move the translations for the other extensions in the other packages.

andrewballantyne reviewed (changes requested):
Looks like you also have a test failure around a CSS variable... `NavHeader.tsx` now imports the CSS file and thus probably lost some context of the global variables it had before. Should probably just be an import in your SCSS file

The error:
```
ERROR in ./public/components/nav/NavHeader.scss
Module build failed (from ./node_modules/mini-css-extract-plugin/dist/loader.js):
ModuleBuildError: Module build failed (from ./node_modules/sass-loader/dist/cjs.js):
SassError: Undefined variable.
  ╷
5 │   @media screen and (min-width: $pf-global--breakpoint--xl) {
  │                                 ^^^^^^^^^^^^^^^^^^^^^^^^^^
  ╵
  public/components/nav/NavHeader.scss 5:33  root stylesheet
    at /go/src/github.com/openshift/console/frontend/node_modules/webpack/lib/NormalModule.js:316:20
    at /go/src/github.com/openshift/console/frontend/node_modules/webpack/node_modules/loader-runner/lib/LoaderRunner.js:367:11
    at /go/src/github.com/openshift/console/frontend/node_modules/webpack/node_modules/loader-runner/lib/LoaderRunner.js:233:18
    at context.callback (/go/src/github.com/openshift/console/frontend/node_modules/webpack/node_modules/loader-runner/lib/LoaderRunner.js:111:13)
    at /go/src/github.com/openshift/console/frontend/node_modules/sass-loader/dist/index.js:62:7
    at Function.call$2 (/go/src/github.com/openshift/console/frontend/node_modules/sass/sass.dart.js:95825:16)
    at _render_closure1.call$2 (/go/src/github.com/openshift/console/frontend/node_modules/sass/sass.dart.js:83882:12)
    at _RootZone.runBinary$3$3 (/go/src/github.com/openshift/console/frontend/node_modules/sass/sass.dart.js:28060:18)
    at _FutureListener.handleError$1 (/go/src/github.com/openshift/console/frontend/node_modules/sass/sass.dart.js:26590:21)
    at _Future__propagateToListeners_handleError.call$0 (/go/src/github.com/openshift/console/frontend/node_modules/sass/sass.dart.js:26889:49)
    at Object._Future__propagateToListeners (/go/src/github.com/openshift/console/frontend/node_modules/sass/sass.dart.js:12505:77)
    at _Future._completeError$2 (/go/src/github.com/openshift/console/frontend/node_modules/sass/sass.dart.js:26735:9)
    at _AsyncAwaitCompleter.completeError$2 (/go/src/github.com/openshift/console/frontend/node_modules/sass/sass.dart.js:26404:12)
    at Object._asyncRethrow (/go/src/github.com/openshift/console/frontend/node_modules/sass/sass.dart.js:12308:17)
    at /go/src/github.com/openshift/console/frontend/node_modules/sass/sass.dart.js:1822:20
    at _wrapJsFunctionForAsync_closure.$protected (/go/src/github.com/openshift/console/frontend/node_modules/sass/sass.dart.js:12333:15)
 @ ./public/components/nav/NavHeader.tsx 20:0-26
 @ ./public/components/nav/index.tsx
 @ ./public/components/app.jsx
```

opayne1 commented:
/label docs-approved

TheRealJon commented:
@andrewballantyne Addressed additional comments and fixed the build error. I'll just need to squash whenever we're ready. @zherman0 If you have any comments to add before I squash, just let me know.

yapei commented:
@TheRealJon tested the changes by updating our console-demo-plugin top level nav menu `Demo Plugin` order with following json
  {
    "type": "console.navigation/section",
    "properties": {
      "id": "admin-demo-section",
      "perspective": "admin",
      "name": "%plugin__console-demo-plugin~Demo Plugin%",
      "insertAfter": "workloads",
      "insertBefore": "networking",
      "dataAttributes": {
        "data-test": "nav-demo-plugin"
      }
    }
  }

`Demo Plugin` nav menu is reordered correctly however I didn't the `data-test` value in `dataAttributes` applied successfully to the element
<img width="350" alt="Screen Shot 2022-05-07 at 19 01 53" src="https://user-images.githubusercontent.com/12692381/167251477-009d3ab4-c86c-4bb6-b262-82374bd9dbc1.png">

TheRealJon commented:
@yapei I found a typo in my original implementation that was causing the data attributes to be committed from dynamic plugin nav section extensions. It should be fixed now.

@andrewballantyne I squashed and rebased. This should be ready for a final review.

andrewballantyne reviewed (approved):
/lgtm

yapei commented:
tested again and no issues found
/label qe-approved

andrewballantyne reviewed (approved):
/lgtm

TheRealJon commented:
/label tide/merge-method-squash

TheRealJon commented:
/retest

andrewballantyne commented:
/lgtm
